app/routes/pages/PageDetailRoute.js

```diff
@@ -109,13 +109,17 @@ const loadData = (props, dispatch) => {
         .concat(dispatch(fetchAllPages()))
     );
   }
+  const itemActions = [];
 
+  for (let i = 0; i < fetchItemActions.length; i++) {
+    itemActions[i] = await dispatch(fetchItemActions[i](pageSlug));
```

Comment:
You can use Promise.all here so that these are fetched concurrently.
```
const itemActions = await Promise.all(  fetchItemActions.map(action => dispatch(action(pageSlug)))  )
```

Reply:
That is hoq it was before. However it is imortant to fetch them in order as the wrong order will lead to an error